matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/DefaultTimeline.kt

```diff
@@ -189,6 +196,11 @@ internal class DefaultTimeline(
                         }
                         .executeBy(taskExecutor)
 
+                if (session.getRoom(roomId)?.isEncrypted().orFalse()
+                        && matrixConfiguration.outboundSessionKeySharingStrategy == OutboundSessionKeySharingStrategy.WhenEnteringRoom) {
```

Comment:
I find it strange to have that here. It's the app that knows when room is entered.
It would be better to have a onEnter() method on Timeline that the app could call

Reply:
Done.